### DISCUSSION TITLE
Is this folder created by djl?

### DISCUSSION THREAD

**Discussion by Guibeen** [deepjavalibrary/djl#3511] [Q&A] [answered]
I noticed this folder in my service environment, and I find it is created during the launching of service.

/home/map is the home directory. I know that DJL downloads and cached pytorch libs by default and .djl.ai/pytorch is the default cache directory, so is this folder the very libraries downloaded?  If the answer is yes, which djl class is doing this operation?
![image](https://github.com/user-attachments/assets/cea69c0b-4d47-4871-952f-35a255fc62dd)

The folloing DJL dependencies are added to the pom.xml, and the jars are packaged to the service libs directory. Is the folder unzipped from the existing jar, pytorch-native-cpu-precxx11-1.13.1-linux-aarch64?
![image](https://github.com/user-attachments/assets/6b9bd79a-38be-46ba-8619-7669cc5d197f)

**Comment by frankfliu** [comment, marked as answer]
@Guibeen 

Yes, the folder is created by DJL, DJL extract shared libraries from jar file:
1. `1.13.1` is the version, in `native/lib/pytorch.properties` file
2. `20221220` is the jar creation time
3. `cpu-precxx11` is the pytorch flavor
4. `linux-aarch64` is the platform

You can find the code here: https://github.com/deepjavalibrary/djl/blob/master/engines/pytorch/pytorch-engine/src/main/java/ai/djl/pytorch/jni/LibUtils.java#L324

**Reply by Guibeen** [reply]
Thank you for your answer! In fact, I'm facing a problem with libstdc++.so.6 in pytorch-native-cpu-precxx11-1.13.1-linux-aarch64.jar: its GLIBCXX version is up to 3.4.19, but my program requires 3.4.21. How can I solve this, is it possible to organize the .so files to a new directory, and add its path to env var PYTORCH_LIBRARY_PATH?

**Comment by frankfliu** [comment]
@Guibeen 
you can set env var or system properties: `PYTORCH_LIBRARY_EXCLUSION=libstdc++.so.6`

It will load your system `libstdc++.so.6` instead of bundled version.

**Reply by Guibeen** [reply]
I have added these commands in the install shell script.
```
export PYTORCH_LIBRARY_EXCLUSION=libstdc++.so.6
export PYTORCH_LIBRARY_PATH=${HOME}/.djl.ai/pytorch/1.13.1-cpu-precxx11-linux-aarch64
export PYTORCH_FLAVOR=cpu-precxx11
```

but it seems not working for PYTORCH_LIBRARY_EXCLUSION. Is there any difference in the processing of these env vars?
![image](https://github.com/user-attachments/assets/3782c541-53bd-44c0-af7c-8a83eb434051)

**Reply by Guibeen** [reply]
it is finally solved by organizing the .so files to a new directory, and adding its path to env var PYTORCH_LIBRARY_PATH
